core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -519,6 +522,9 @@ public void headersRead(Metadata headers) {
       commitAndRun(substream);
       if (state.winningSubstream == substream) {
         masterListener.headersRead(headers);
+        if (throttle != null) {
+          throttle.onSuccess();
```

Comment:
Isn't this too early to say the RPC is a success? What if the winning substream later fails with a retryable status code? I think this would ends up with the same stream recording as a success (increment) and a failure (decrement) to the throttling policy's token count.

Reply:
I missed that committing the substream here will prevent the `#makeRetryDecision()` code from running when the stream is closed, so it can't subsequently decrement the count.